src/test/java/com/training/web/tests/FlipkartTest.java

```diff
@@ -224,38 +233,36 @@ public void testProductPageDesign(String item, int position) {
         ProductPage productPage = new HomePage().popUpCancel().setShoes(item).searchShoes().OpenProductPage(position);
         productPage.scrollToBottom();
         ArrayList<String> headers = new ArrayList<>(Arrays.asList(map.get("headers").split(",")));
-        for (String headersInSelectionPage : headers) {
-            softAssert.assertThat(productPage.isSelectionDisplayed(headersInSelectionPage)).
-                    as("Json selected Headers is Not matching with selection page Headers").isTrue();
 
-        }
+        headers.stream().forEach(header ->
+                softAssert.assertThat(productPage.isSelectionDisplayed(header)).
+                        as("Json selected Headers is Not matching with selection page Headers").isTrue());
     }
 
     /**
      * flipkartSocialMedia(): Clicking the social media links and checking the new tab url and comparing it with links present in json
      */
-
     @Test
-    public void flipkartSocialMedia(){
-        int counter=0;
+    public void flipkartSocialMedia() {
+        AtomicInteger counter = new AtomicInteger();
         HomePage homePage = new HomePage().popUpCancel();
-        SocialMediaPage socialMediaPage=null;
-        ArrayList<String> socialMediaLinksArray = new ArrayList<>(Arrays.asList(map.get("links")
+
+        ArrayList<String> socialMediaLinksArrayList = new ArrayList<>(Arrays.asList(map.get("links")
                 .split(",")));
-        ArrayList<String> placeholdersOfSocialSites= new ArrayList<>(Arrays.asList(map.get("placeholders")
+        ArrayList<String> placeholdersOfSocialSites = new ArrayList<>(Arrays.asList(map.get("placeholders")
                 .split(",")));
-        for (String socialMediaLinks:socialMediaLinksArray) {
-            logger.info("*****links*****"+socialMediaLinks);
-            logger.info("*****placeholders*****"+placeholdersOfSocialSites);
-            socialMediaPage =homePage.clickLink(socialMediaLinks);
-            String url = socialMediaPage.getSocialMediaUrl(placeholdersOfSocialSites.get(counter));
-            logger.info("********" + url);
-            logger.info("********" + socialMediaLinks.toLowerCase());
-            softAssert.assertThat(url).contains(socialMediaLinks.toLowerCase()).
+
+        socialMediaLinksArrayList.stream().forEach(socialMediaLink -> {
```

Comment:
Whenever you want to play with indexes - always use `IntStream.range()` or `rangeClosed()`
Then you would need to define a counter.